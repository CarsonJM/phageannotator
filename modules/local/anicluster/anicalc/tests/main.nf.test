nextflow_process {

    name "Test process: ANICLUSTER_ANICALC"
    script "../main.nf"
    process "ANICLUSTER_ANICALC"

    test("blast.txt") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.test_data['modules_local']['blast_txt'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("blast.txt - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.test_data['modules_local']['blast_txt'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
