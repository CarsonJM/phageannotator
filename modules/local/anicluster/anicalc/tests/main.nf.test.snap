{
    "['modules_local']['blast_txt']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,1934eb09849f6df88083ad25aa47923d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ],
                "ani": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,1934eb09849f6df88083ad25aa47923d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ]
            }
        ],
        "timestamp": "2023-11-13T11:49:17.964318426"
    }
}